src/app/backend/replicationcontrollerdetail.go

```diff
@@ -322,6 +317,37 @@ func getInternalEndpoint(serviceName, namespace string, ports []api.ServicePort)
 	}
 }
 
+// Returns array of external endpoints for a replication controller.
+func getExternalEndpoints(replicationController api.ReplicationController, pods []api.Pod,
+	service api.Service) []Endpoint {
+	var externalEndpoints []Endpoint
+	for _, externalIP := range service.Status.LoadBalancer.Ingress {
+		externalEndpoints = append(externalEndpoints, getExternalEndpoint(externalIP, service.Spec.Ports))
+	}
+	// For type node ports or if load balancer is unavailable display node ports.
+	if (len(externalEndpoints) <= 0 && service.Spec.Type == api.ServiceTypeLoadBalancer) ||
+		service.Spec.Type == api.ServiceTypeNodePort {
+		for _, pod := range pods {
+			if isLabelSelectorMatching(replicationController.Spec.Selector, pod.Labels) {
+				for _, port := range service.Spec.Ports {
+					externalEndpoints = append(externalEndpoints, Endpoint{
+						Host: pod.Status.HostIP,
+						Ports: []ServicePort{
+							{
+								Protocol: port.Protocol,
+								Port:     port.NodePort,
+							},
+						},
+					})
+				}
+				// Returning external endpoints for only one pod.
```

Comment:
Actually the list of external enpoints is not that long shen using node IP instead of pod IP, so I think we should display all of them. It's already done, please chceck code/test how it works for you.